src/ngraph/op/sum.cpp

```diff
@@ -20,27 +20,61 @@
 using namespace std;
 using namespace ngraph;
 
-const string op::Sum::type_name{"Sum"};
+const string op::v0::Sum::type_name{"Sum"};
 
-op::Sum::Sum(const Output<Node>& arg, const AxisSet& reduction_axes)
+op::v0::Sum::Sum(const Output<Node>& arg, const AxisSet& reduction_axes)
     : ArithmeticReduction(arg, reduction_axes)
 {
     constructor_validate_and_infer_types();
 }
 
-op::Sum::Sum(const Output<Node>& arg, const Output<Node>& reduction_axes)
+op::v0::Sum::Sum(const Output<Node>& arg, const Output<Node>& reduction_axes)
     : ArithmeticReduction(arg, reduction_axes)
 {
     constructor_validate_and_infer_types();
 }
 
-shared_ptr<Node> op::Sum::copy_with_new_args(const NodeVector& new_args) const
+shared_ptr<Node> op::v0::Sum::copy_with_new_args(const NodeVector& new_args) const
 {
     check_new_args_count(this, new_args);
     return make_shared<Sum>(new_args.at(0), new_args.at(1));
 }
 
-void op::Sum::generate_adjoints(autodiff::Adjoints& adjoints, const NodeVector& deltas)
+void op::v0::Sum::generate_adjoints(autodiff::Adjoints& adjoints, const NodeVector& deltas)
+{
+    auto delta = deltas.at(0);
+
+    auto x = input_value(0);
+    auto& x_shape = x.get_shape();
+
+    adjoints.add_delta(x, make_shared<op::Broadcast>(delta, x_shape, get_reduction_axes()));
+}
+
+const string op::v1::ReduceSum::type_name{"Sum"};
```

Comment:
```suggestion
const string op::v1::ReduceSum::type_name{"ReduceSum"};
```

Reply:
Done. I think about it, the side effect is lack of support by `get_typeid` (in this moment it is not used). Probably is will be resolved during [Remove static initializers from ops](https://github.com/NervanaSystems/ngraph/pull/3533/files)

Reply:
Let's defer until later, maybe as part of a serializer cleanup.